tools/operator/launcher/k8s/createStatefulset.go

```diff
@@ -0,0 +1,377 @@
+	if err != nil {
+		return err
+	}
+
+	var replicas int32 = 1
+	statefulsetClient := clientset.AppsV1().StatefulSets(ns)
+	statefulsetRes := &apiv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: launchConfig.Name,
+		},
+		Spec: apiv1.StatefulSetSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"k8s-app": launchConfig.Name,
+					"type":    launchConfig.Type,
+				},
+			},
+			ServiceName: launchConfig.Name,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"k8s-app": launchConfig.Name,
+						"type":    launchConfig.Type,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Volumes:    launchConfig.Volumes,
+					Containers: launchConfig.Containers,
+				},
+			},
+		},
+	}
+
+	if nsConfig.Metrics {
+		annotations := map[string]string{
+			"prometheus.io/scrape": "true",
+			"prometheus.io/path":   "metrics",
+			"prometheus.io/port":   "9443",
+			"prometheus.io/scheme": "http",
+		}
+		statefulsetRes.Spec.Template.ObjectMeta.Annotations = annotations
+	}
+
+	logger.INFO("Creating Statefulset for ", launchConfig.Name)
+	result, err := statefulsetClient.Create(statefulsetRes)
+	if err != nil {
+		return errors.Wrap(err, "failed to create statefulset")
+	}
+	logger.INFO("Created Statefulsets ", result.GetName())
+	return nil
+}
+
+//CreateConfigMap --
+func (k8s K8s) CreateConfigMap(name, componentType, certsType, namespace string, nsConfig networkspec.Config, clientset *kubernetes.Clientset) error {
+
+	configmapRes := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-%s", name, certsType),
+		},
+		Data: k8s.certsLists(name, componentType, certsType, nsConfig),
+	}
+
+	_, err := clientset.CoreV1().ConfigMaps(namespace).Create(configmapRes)
+	if err != nil {
+		return errors.Wrap(err, "failed to create configmap")
+	}
+	return nil
+}
+
+//CreateNameSpace --
+func (k8s K8s) CreateNameSpace(ns string, clientset *kubernetes.Clientset) error {
+
+	nsSpec := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns}}
+	_, err := clientset.CoreV1().Namespaces().Create(nsSpec)
+	if err != nil {
+		return errors.Wrap(err, "failed to create namespace")
+	}
+	return nil
+}
+
+//DeleteNameSpace --
+func (k8s K8s) DeleteNameSpace(ns string, clientset *kubernetes.Clientset) error {
+
+	var gracePeriodSeconds int64 = 0
+	var deletionPropagation metav1.DeletionPropagation = "Background"
+	options := &metav1.DeleteOptions{GracePeriodSeconds: &gracePeriodSeconds, PropagationPolicy: &deletionPropagation}
+	err := clientset.CoreV1().Namespaces().Delete(ns, options)
+	if err != nil {
+		return errors.Wrap(err, "failed to delete namespace")
+	}
+	status := make(chan string, 1)
+	go func() {
+		for {
+			_, err = clientset.CoreV1().Namespaces().Get(ns, metav1.GetOptions{})
+			if err != nil {
+				status <- "result"
+			}
+			time.Sleep(30 * time.Second)
```

Comment:
Can we reduce this to 10 seconds, while running tests quickly back to back, I don't want to wait 30 seconds at the end of every test where 10 seconds may have sufficed.